lib/src/firebase_storage_mocks_base.dart

```diff
@@ -9,26 +11,76 @@ class MockFirebaseStorage extends Mock implements FirebaseStorage {
   final Map<String, Uint8List> storedDataMap = {};
 
   @override
-  StorageReference ref() {
-    return MockStorageReference(this);
+  Reference ref([String? path]) {
+    path ??= '/';
+    return MockReference(this, path);
   }
+
+  @override
+  String get bucket => 'some-bucket';
 }
 
-class MockStorageUploadTask extends Mock implements StorageUploadTask {
-  final StorageReference reference;
+class MockUploadTask extends Mock implements UploadTask {
+  final Future<TaskSnapshot> delegate;
+  final TaskSnapshot _snapshot;
+
+  MockUploadTask(reference)
+      : delegate = Future.value(MockTaskSnapshot(reference)),
+        _snapshot = MockTaskSnapshot(reference);
+
+  @override
+  Future<S> then<S>(FutureOr<S> Function(TaskSnapshot) onValue,
+          {Function? onError}) =>
+      delegate.then(onValue, onError: onError);
+
+  @override
+  Stream<TaskSnapshot> asStream() {
+    return delegate.asStream();
+  }
 
-  MockStorageUploadTask(this.reference);
+  @override
+  Future<TaskSnapshot> whenComplete(Function action) {
+    return delegate;
+  }
+
+  @override
+  Future<TaskSnapshot> timeout(Duration timeLimit,
+      {FutureOr<TaskSnapshot> Function()? onTimeout}) {
+    return delegate.timeout(timeLimit);
```

Comment:
Would you mind also passing onTimout to the delegate?

https://api.dart.dev/stable/2.10.5/dart-async/Future/timeout.html